src/dbos-executor.ts

```diff
@@ -345,8 +345,16 @@ export class DBOSExecutor implements DBOSExecutorContext {
         await this.recoverPendingWorkflows();
       }
     } catch (err) {
-      (err as Error).message = `failed to initialize workflow executor: ${(err as Error).message}`;
-      throw new DBOSInitializationError(`${(err as Error).message}`);
+      if (err instanceof AggregateError) {
+        let combinedMessage = 'Failed to initialize workflow executor: ';
+        for (const error of err.errors) {
+          combinedMessage += `${(error as Error).message}; `;
+        }
+        throw new DBOSInitializationError(combinedMessage);
+      } else if (err instanceof Error) {
```

Comment:
What if `err` is not an instance of `Error`? We should still throw an error no matter what.